agent/recorder.go

```diff
@@ -149,7 +150,18 @@ func (r *SpanRecorder) SendSpans() (error, bool) {
 		}
 		statusCode, err := r.callIngest(buf)
 		if err != nil {
-			if statusCode == 401 {
+			if statusCode == 0 {
+				// To handle client timeout errors (like TLS handshake timeout)
+				switch erType := err.(type) {
+				case interface{ Timeout() bool }:
+					if erType.Timeout() {
+						lastError = err
+						r.logger.Printf("error: client timeout, retrying (%d) in %d seconds",
+							i, retryBackoff/time.Second)
+						time.Sleep(retryBackoff)
+					}
+				}
```

Comment:
I think in general, any connection error should be retried, right?

Reply:
There are some errors that are interesting, like the ones with timeout:

![image](https://user-images.githubusercontent.com/69803/74227454-ca509f80-4cbe-11ea-80b0-a85919dd2953.png)

but there's many other that I think is not valid for retry:

![image](https://user-images.githubusercontent.com/69803/74227516-e9e7c800-4cbe-11ea-903f-27b89dbd96b8.png)
![image](https://user-images.githubusercontent.com/69803/74227560-0257e280-4cbf-11ea-92d7-5ef96d9b42bc.png)
![image](https://user-images.githubusercontent.com/69803/74227592-16034900-4cbf-11ea-9f9b-c3abdafe8dcb.png)

The client you send me have the following policy:
https://github.com/hashicorp/go-retryablehttp/blob/master/client.go#L363-L400

Right now we retry on timeouts and on status codes >= 500

Do you want me to change and retry for all? to match the policy of `go-retryablehttp`? or leave it like what we have?

Reply:
I'd retry for all. Most of those errors can never happen, and if they do, retrying 3 times should be quick and painless. We can copy the policy of `go-retryablehttp` which includes most errors.

Reply:
Ok, I will copy the policy.